### PR TITLE
Do not build PRs of non-members

### PR DESCRIPTION
**Pull request by mrks**
While Jenkins is configured to not accept new `Jenkinsfile`s from untrusted sources, the code will still be build. Even without a concrete threat scenario, this does not seem to be a good idea.

From now on, a maintainer will have to add new users to either [Zweirise-Developers](https://github.com/orgs/hyrise/teams/zweirise-developers/members) (for Hiwis, master's theses, ... - includes write access to the repo) or [Seminar](https://github.com/orgs/hyrise/teams/seminar/members) (for students in the seminar - no direct write access).

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -42,7 +59,7 @@ node {
         mkdir gcc-debug && cd gcc-debug && cmake -DCI_BUILD=ON -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ .. &\
         mkdir gcc-release && cd gcc-release && cmake -DCI_BUILD=ON -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ .. &\
         wait"
-        full_ci = sh(script: "./scripts/current_branch_has_pull_request_label.py FullCI", returnStdout: true).trim() == "true"
+        full_ci = pullRequest.labels.contains('FullCI')
```
Comment (Bensk1) [inline Jenkinsfile]: Is this enabled by a new version of Jenkins?

File: Jenkinsfile [hunk 2]
```diff
@@ -2,6 +2,23 @@ import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
 
 node {
   stage ("Start") {
+    // Check if the user who opened the PR is a known collaborator (i.e., has been added to a hyrise/hyrise team)
+    try {
+      withCredentials([usernamePassword(credentialsId: '5fe8ede9-bbdb-4803-a307-6924d4b4d9b5', usernameVariable: 'GITHUB_USERNAME', passwordVariable: 'GITHUB_TOKEN')]) {
```
Comment (Bensk1) [inline Jenkinsfile]: Is `GITHUB_TOKEN` set in the Jenkins settings, e.g., via the web interface?

Comment (mrks) [reply Jenkinsfile]: No, it is set right here. We pass in the name of a variable as a string. A superior concept, which should definitely find its way into C++!

File: Jenkinsfile (hunk 1 above)
Comment (mrks) [reply Jenkinsfile]: No, by me discovering the github pipeline plugin.

**Pull request merged**